text/0000-process-specific-working-directory.md

```diff
@@ -0,0 +1,65 @@
+# Meta
+[meta]: #meta
+- Name: Process Specific Working Directory
+- Start Date: 2021-03-09
+- Author(s): ForestEckhardt, ryanmoran
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+For any given process, the buildpack should be able to specify a working
+directory that is not `CNB_APP_DIR`.
+
+# Motivation
+[motivation]: #motivation
+
+Allowing a buildpack to specify a custom working directory would allow
+buildpacks to write start commands that need to be invoked from a specfic
+directory without needing to invoke a shell.
+
+# What it is
+[what-it-is]: #what-it-is
+
+It is a common pattern in buildpacks to execute the process from a directory
+that is not `CNB_APP_DIR`. In these cases, buildpack authors are required to
+write a process command that includes a change into the directory that the
+process should run from. This often looks like `cd $CNB_APP_DIR/sub-directory &&
+./execute-process`. Unfortunately, this means that the command can only be run
+on a stack that includes a shell. It also means that including additional flags
+or arguments to the command at runtime is cumbersome.
+
+As an alternative, the buildpack lifecycle could allow buildpacks to specify
+the working directory of the process when it is invoked.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+A new field `directory` would be added the the `processes` in `launch.toml`.
+```toml
+[[processes]]
+type = "<process type>"
+command = "<command>"
+args = ["<arguments>"]
+direct = false
+directory = "<working directory>"
+```
+
+If `directory` is not set the launcher will be invoked in `CNB_APP_DIR` as
```

Comment:
Summarizing from working group, might be good to clarify whether the change in working directory is intended to affect profile.d / exec.d for a given process.

Reply:
Also it's worth noting that profile.d and exec.d can also exist __per process__ which might get a little confusing if the working directory for the process is not the same.

One idea that came up in implementation sub-team sync: only doing this for direct processes. (Since for non-direct processes, you could still `cd` into the directory at runtime).

Reply:
@ForestEckhardt and I were chatting and we thought it'd be sort of confusing if the `working-directory` specified in the `launch.toml` impacted the working dir for profile.d and exec.d scripts. Reading the `launch.toml`, for me the most intuitive interpretation is that the `working-directory` applies to the `command` in the way that `args` also applies to the `command`. (Notably, `args` don't get passed to the profile.d or exec.d scripts.)

If a profile.d or exec.d scripts need to run in a certain working dir, they can `cd` in there at the start of their runs.

Reply:
Though I haven't yet authored a buildpack that generates a profile.d or exec.d script. Would be interested to hear from buildpack authors about what makes sense/is convenient for them.

Reply:
I'd agree with @fg-j's expectation for how `working-directory` would apply to a `profile.d` or `exec.d` script. In both cases they can be written to change their own working directory. We don't have the same kind of support for a process to change its working directory outside of invoking a shell command to do so.